pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -324,37 +324,12 @@ private boolean registerNamespace(String namespace, boolean ensureOwned) throws
             targetMap = findingBundlesNotAuthoritative;
         }
 
-        return targetMap.computeIfAbsent(bundle, (k) -> {
-            CompletableFuture<Optional<LookupResult>> future = new CompletableFuture<>();
-
-            // First check if we or someone else already owns the bundle
-            ownershipCache.getOwnerAsync(bundle).thenAccept(nsData -> {
-                if (!nsData.isPresent()) {
-                    // No one owns this bundle
+        if (targetMap.get(bundle) != null && !targetMap.get(bundle).isDone()) {
```

Comment:
the two get calls can *potentially* return different values. so as a good practice, I would recommend writing in the following way.

```
CompletableFuture<Optional<LookupResult>> future = targetMap.get(bundle);
if (future != null && !future.isDone()) {
    // ...
}
```